### ISSUE TITLE
ping host.net_default is instant, ping host takes 5 seconds consistently

### ISSUE THREAD

**Issue by marcelmfs** [docker/docker#21126]
**Output of `docker version`:**

```
Client:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   c3959b1
 Built:        Mon Feb 22 22:37:33 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   c3959b1
 Built:        Mon Feb 22 22:37:33 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 38
 Running: 20
 Paused: 0
 Stopped: 18
Images: 30
Server Version: 1.10.2
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 337
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Plugins:
 Volume: local
 Network: null host bridge
Kernel Version: 4.1.18-boot2docker
Operating System: Boot2Docker 1.10.2 (TCL 6.4.1); master : 611be10 - Mon Feb 22 22:47:06 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 7.79 GiB
Name: default
ID: PCLY:RFNJ:JDGC:4XUP:CKTQ:LROS:SH7R:ZONH:JUDN:ELTN:P32W:NE7P
Debug mode (server): true
 File Descriptors: 226
 Goroutines: 296
 System Time: 2016-03-11T16:10:21.273403804Z
 EventsListeners: 0
 Init SHA1:
 Init Path: /usr/local/bin/docker
 Docker Root Dir: /mnt/sda1/var/lib/docker
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

- All containers are using alpine:3.3
- The default network for the case of this report is net_default
- There are some containers in the same network

**Steps to reproduce the issue:**
1. `docker run --net net_default -it alpine:3.3 /bin/sh`
2. 
```
/ # time ping -c 1 mongodb
PING mongodb (172.18.0.29): 56 data bytes
64 bytes from 172.18.0.29: seq=0 ttl=64 time=0.302 ms

--- mongodb ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 0.302/0.302/0.302 ms
real	0m 5.00s
user	0m 0.00s
sys	0m 0.00s
/ # time ping -c 1 mongodb.net_default
PING mongodb.net_default (172.18.0.29): 56 data bytes
64 bytes from 172.18.0.29: seq=0 ttl=64 time=0.068 ms

--- mongodb.net_default ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 0.068/0.068/0.068 ms
real	0m 0.00s
user	0m 0.00s
sys	0m 0.00s
/ # time nslookup mongodb
nslookup: can't resolve '(null)': Name does not resolve

Name:      mongodb
Address 1: 172.18.0.29 mongodb.net_default
real	0m 5.00s
user	0m 0.00s
sys	0m 0.00s
/ # time nslookup mongodb.net_default
nslookup: can't resolve '(null)': Name does not resolve

Name:      mongodb.net_default
Address 1: 172.18.0.29 mongodb.net_default
real	0m 0.00s
user	0m 0.00s
sys	0m 0.00s
/ #
```

**Describe the results you received:**

- 5 seconds response time ping without `.net_default` **tld**

**Describe the results you expected:**

- 0 seconds response time ping without `.net_default` **tld**

**Comment by mavenugo** [comment]
@marcelmfs PTAL https://github.com/docker/docker/issues/20335#issuecomment-190291821. This is most likely the v6 resolution issue. Can you try `ping -4 xxxxxxx` to confirm if there is any delay when forcing a v4 only resolution ?

**Comment by marcelmfs** [comment]
Yes, I knew it was the ipv6/ipv4 issue, what I don't understand is why adding the complete `localhost.localdomain` "FQDN" works for ipv4 and not for ipv6. Is there any workaround in terms of the host vm configuration or daemon startup parametrisation to mitigate that behaviour on docker container regarding to name resolution?

**Comment by marcelmfs** [comment]
I can confirm that docker 1.11 fixes this issue. Will do some tests with sysctl as https://github.com/docker/docker/pull/19265 was merged to also report on that.

**Comment by thaJeztah** [comment]
Thanks for testing @marcelmfs, I'll close this issue, because the issue reported here seems to be resolved in 1.11 (https://github.com/docker/docker/pull/19265 will be in 1.12)
